libr/bin/p/bin_bflt.c

```diff
@@ -181,22 +180,17 @@ static RList *relocs(RBinFile *bf) {
 	}
 
 	if (obj->hdr->reloc_count > 0) {
-		int n_reloc = obj->hdr->reloc_count;
-
-		amount = n_reloc * sizeof (struct reloc_struct_t);
-		if (amount < n_reloc || amount > UT32_MAX) {
+		ut32 n_reloc = obj->hdr->reloc_count;
+		if (n_reloc > UT32_MAX / sizeof (struct reloc_struct_t)) {
 			goto out_error;
 		}
-		struct reloc_struct_t *reloc_table = calloc (1, amount + 1);
+		amount = n_reloc * sizeof (struct reloc_struct_t);
+		struct reloc_struct_t *reloc_table = calloc (1, amount);
 		if (!reloc_table) {
 			goto out_error;
 		}
 		amount = n_reloc * sizeof (ut32);
-		if (amount < n_reloc || amount > UT32_MAX) {
-			free (reloc_table);
-			goto out_error;
-		}
-		ut32 *reloc_pointer_table = calloc (1, amount + 1);
+		ut32 *reloc_pointer_table = calloc (1, amount);
```

Comment:
amount can be 0, so this will be UB

Reply:
```c
if (obj->hdr->reloc_count > 0) {
		ut32 n_reloc = obj->hdr->reloc_count;
		if (n_reloc > UT32_MAX / sizeof (struct reloc_struct_t)) {
			goto out_error;
		}
		amount = n_reloc * sizeof (struct reloc_struct_t);
```
Here n_reloc>=1, so amount can't be zero.